init-gateway issue on Windows nodes after reboot

After Windows node reboot the ovn-kubernetes-node service is Stopped. Error:

```
time="2018-11-29T17:01:10Z" level=info msg="Node atv-win3 ready for ovn initialization with subnet 10.0.1.0/24"
time="2018-11-29T17:01:25Z" level=error msg="failed to init shared interface gateway: Failed to add logical port to router, stdout: \"\", stderr: \"ovn-nbctl: rtoe-GR_atv-win3: port already exists with mac fa:16:3e:7d:1e:e6\\r\\n\", error: exit status 1"
panic: failed to init shared interface gateway: Failed to add logical port to router, stdout: "", stderr: "ovn-nbctl: rtoe-GR_atv-win3: port already exists with mac fa:16:3e:7d:1e:e6\r\n", error: exit status 1

goroutine 1 [running]:
main.runOvnKube(0xc04222f760, 0x0, 0x0)
	/tmp/ovn-kubernetes-checkout/go-controller/_output/go/src/github.com/openvswitch/ovn-kubernetes/go-controller/cmd/ovnkube/ovnkube.go:259 +0x1a51
main.main.func1(0xc04222f760, 0xc04222f760, 0xc0424ff7ff)
	/tmp/ovn-kubernetes-checkout/go-controller/_output/go/src/github.com/openvswitch/ovn-kubernetes/go-controller/cmd/ovnkube/ovnkube.go:117 +0x32
github.com/openvswitch/ovn-kubernetes/go-controller/vendor/github.com/urfave/cli.HandleAction(0x1135880, 0x13589b8, 0xc04222f760, 0xc0420c5380, 0x0)
	/tmp/ovn-kubernetes-checkout/go-controller/_output/go/src/github.com/openvswitch/ovn-kubernetes/go-controller/vendor/github.com/urfave/cli/app.go:501 +0xd9
github.com/openvswitch/ovn-kubernetes/go-controller/vendor/github.com/urfave/cli.(*App).Run(0xc04236a700, 0xc042084000, 0x18, 0x20, 0x0, 0x0)
	/tmp/ovn-kubernetes-checkout/go-controller/_output/go/src/github.com/openvswitch/ovn-kubernetes/go-controller/vendor/github.com/urfave/cli/app.go:268 +0x65a
main.main()
	/tmp/ovn-kubernetes-checkout/go-controller/_output/go/src/github.com/openvswitch/ovn-kubernetes/go-controller/cmd/ovnkube/ovnkube.go:120 +0xac5

```

The deployed cluster worked just fine until the reboot.

--- a/go-controller/pkg/ovn/management-port.go
+++ b/go-controller/pkg/ovn/management-port.go
@@ -3,7 +3,6 @@ package ovn
 import (
 	"fmt"
 	"net"
-	"os/exec"
 	"runtime"
 	"strconv"
 	"strings"
@@ -283,15 +282,10 @@ func CreateManagementPort(nodeName, localSubnet,
 	}
 
 	if runtime.GOOS == windowsOS && macAddress == "00:00:00:00:00:00" {
-		var stdoutStderr []byte
-		stdoutStderr, err = exec.Command("powershell", "$(Get-NetAdapter", "-IncludeHidden", "|", "Where", "{", "$_.Name",
-			"-Match", fmt.Sprintf("\"%s\"", interfaceName), "}).MacAddress").CombinedOutput()
+		macAddress, err = util.FetchIfMacWindows(interfaceName)
 		if err != nil {
-			logrus.Errorf("Failed to get mac address of ovn-k8s-master, stderr: %q, error: %v", fmt.Sprintf("%s", stdoutStderr), err)
 			return err
 		}
-		// Windows returns it in 00-00-00-00-00-00 format, we want ':' instead of '-'
-		macAddress = strings.Replace(strings.TrimSpace(fmt.Sprintf("%s", stdoutStderr)), "-", ":", -1)
 	}
 
 	// Create the OVN logical port.

--- a/go-controller/pkg/util/gateway-init.go
+++ b/go-controller/pkg/util/gateway-init.go
@@ -4,6 +4,7 @@ import (
 	"fmt"
 	"github.com/sirupsen/logrus"
 	"net"
+	"runtime"
 	"strings"
 )
 
@@ -318,6 +319,12 @@ func GatewayInit(clusterIPSubnet []string, nodeName, nicIP, physicalInterface,
 		}
 		if macAddress == "" {
 			return fmt.Errorf("No mac_address found for the bridge-interface")
+		}
+		if runtime.GOOS == windowsOS && macAddress == "00:00:00:00:00:00" {
+			macAddress, err = FetchIfMacWindows(bridgeInterface)
+			if err != nil {
+				return err
+			}
 		}
 		stdout, stderr, err = RunOVSVsctl("set", "bridge",
 			bridgeInterface, "other-config:hwaddr="+macAddress)

--- a/go-controller/pkg/util/util.go
+++ b/go-controller/pkg/util/util.go
@@ -2,6 +2,9 @@ package util
 
 import (
 	"fmt"
+	"os/exec"
+	"strings"
+
 	"github.com/urfave/cli"
 )
 
@@ -12,4 +15,19 @@ func StringArg(context *cli.Context, name string) (string, error) {
 		return "", fmt.Errorf("argument --%s should be non-null", name)
 	}
 	return val, nil
+}
+
+// FetchIfMacWindows gets the mac of the interfaceName via powershell commands
+// There is a known issue with OVS not correctly picking up the
+// physical network interface MAC address.
+func FetchIfMacWindows(interfaceName string) (string, error) {
+	stdoutStderr, err := exec.Command("powershell", "$(Get-NetAdapter", "-IncludeHidden",
+		"-InterfaceAlias", fmt.Sprintf("\"%s\"", interfaceName), ").MacAddress").CombinedOutput()
+	if err != nil {
+		return "", fmt.Errorf("Failed to get mac address of ovn-k8s-master, stderr: %q, error: %v", fmt.Sprintf("%s", stdoutStderr), err)
+	}
+	// Windows returns it in 00-00-00-00-00-00 format, we want ':' instead of '-'
+	macAddress := strings.Replace(strings.TrimSpace(fmt.Sprintf("%s", stdoutStderr)), "-", ":", -1)
+
+	return strings.ToLower(macAddress), nil
 }

